auto_nag/scripts/prod_comp_changed_with_priority.py

```diff
@@ -0,0 +1,90 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import dateutil.parser
+from libmozdata.bugzilla import Bugzilla
+from auto_nag.bzcleaner import BzCleaner
+
+
+class ProdCompChangedWithPriority(BzCleaner):
+    def __init__(self):
+        super(ProdCompChangedWithPriority, self).__init__()
+
+    def description(self):
+        return 'Bugs with a priority set before product::component changed'
+
+    def filter_bugs(self, bugs):
+        def history_handler(bug, data):
+            priority_date = None
+            prod_comp_date = None
+            priority_who = None
+            prod_comp_who = None
+            for h in bug['history']:
+                for change in h['changes']:
+                    if change['field_name'] == 'priority':
+                        priority_date = dateutil.parser.parse(h['when'])
+                        priority_who = h['who']
+                    elif change['field_name'] in {'product', 'component'}:
+                        prod_comp_date = dateutil.parser.parse(h['when'])
+                        prod_comp_who = h['who']
+            if (
+                priority_date is None
+                or prod_comp_date is None
+                or priority_date >= prod_comp_date
+                or priority_who == prod_comp_who
```

Comment:
Maybe we can avoid resetting the priority when the triage owner is the same as the one who set the priority?

Reply:
This way we will also avoid resetting the priority when the component before/after is in the same team

Reply:
yeah, I thought about that too... but I'm not sure we can rely on people memory. I think a prod::comp change requires a reevaluation of the priority.
Anyway, I'm not sure...
@sylvestre wdyt ?